populatingNextRightPointersInEachNode.py

```diff
@@ -0,0 +1,28 @@
+# Definition for binary tree with next pointer.
+# class TreeLinkNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+#         self.next = None
+
+class Solution:
+    # @param root, a tree link node
+    # @return nothing
+    def connect(self, root):
+        queue = []
+        if root:
+            queue.append(root)
+        while queue:
+            newQueue = []
+            while queue:
+                node = queue.pop(0)
```

Comment:
should probably use deque (https://docs.python.org/2/library/collections.html#collections.deque)
so that both `pop(0)` and `pop()` has O(1) time complexity.

Reply:
Is time complexity larger than O(1) using array to do pop(0) and pop()?

Reply:
Array `pop(0)` is O(N)...